spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -402,9 +404,13 @@ def measure(self):
             printv('\nMeasures on lesion #' + str(lesion_label) + '...', self.verbose, 'normal')
 
             label_idx = self.measure_pd[self.measure_pd.label == lesion_label].index
-            self._measure_volume(im_lesion_data_cur, p_lst, label_idx)
-            self._measure_length(im_lesion_data_cur, p_lst, label_idx)
-            self._measure_diameter(im_lesion_data_cur, p_lst, label_idx)
+            # No angle correction is needed for the volume measurement
+            if self.fname_sc is None:
+                self._measure_volume(im_lesion_data_cur, p_lst, label_idx)
+            else:
+                self._measure_volume(im_lesion_data_cur, p_lst, label_idx)
+                self._measure_length(im_lesion_data_cur, p_lst, label_idx)
+                self._measure_diameter(im_lesion_data_cur, p_lst, label_idx)
```

Comment:
Very small nitpick, but I would swap the two cases, that way we match the other `is not None` check above:

```suggestion
            # If SC segmentation is provided, we can compute length and diameter
            if self.fname_sc is not None:
                self._measure_volume(im_lesion_data_cur, p_lst, label_idx)
                self._measure_length(im_lesion_data_cur, p_lst, label_idx)
                self._measure_diameter(im_lesion_data_cur, p_lst, label_idx)
            # Otherwise, without SC seg, we can only compute the volume
            else:
                self._measure_volume(im_lesion_data_cur, p_lst, label_idx)
```

Reply:
We may even want to simplify this further so that there's only a single call to `self._measure_volume`:

```suggestion
            # If SC segmentation is provided, we can compute length, diameter, and volume
            # Otherwise, without SC seg, we can only compute the volume
            if self.fname_sc is not None:
                self._measure_length(im_lesion_data_cur, p_lst, label_idx)
                self._measure_diameter(im_lesion_data_cur, p_lst, label_idx)
            self._measure_volume(im_lesion_data_cur, p_lst, label_idx)
```

But, I'll leave it up to you as to which one you prefer, as I'm okay with either. :slightly_smiling_face: